### PR TITLE
Creates Initial component

### PR DESCRIPTION
**Pull request by shaneriley**
This component renders an array of bookmark items and emits events when a bookmark is clicked and when an edit action for a bookmark is clicked. Completes [WEBDEV-3814](https://webarchive.jira.com/browse/WEBDEV-3814).

bookmarks-list

### CODE REVIEW

**Review by bfalling — APPROVED** [review]
LGTM

File: src/styles/ia-bookmarks-list.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+p {
+  padding: 0;
+  margin: 0;
+  grid-column: 1 / 4;
+}
+
+button {
+  -webkit-appearance: none;
+  appearance: none;
+  width: 20px;
```
Comment (iisa) [inline src/styles/ia-bookmarks-list.js:49]: please use rem where you can

File: src/styles/ia-bookmarks-list.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  padding: 0 1rem;
+  color: var(--primaryTextColor);
+}
+
+small {
+  font-style: italic;
+}
+
+ul {
+  padding: 0;
```
Comment (iisa) [inline src/styles/ia-bookmarks-list.js]: please make sure that scrolling to bottom allows to reach furthest text
```suggestion
  padding: 0 0 2rem 0;
```

File: src/ia-bookmarks-list.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    super();
+    this.activeBookmarkID = undefined;
+    this.bookmarks = [];
+    this.renderHeader = false;
+  }
+
+  emitEditEvent(e, bookmark) {
+    e.stopPropagation();
+    this.dispatchEvent(new CustomEvent('bookmarkEdited', {
+      bubbles: true,
```
Comment (iisa) [inline src/ia-bookmarks-list.js]: if we are catching these events targeting its node, then we do not need to bubble further up

**Review by iisa — CHANGES_REQUESTED** [review]
some cleanup.
what I am seeing is that these submenus have a hard style on them that may cut off content at the bottom.
Unsure what it is here but we need to move towards a "graceful cascade" - where the menus grow however larger they get and that the parent container allows for the overflow & can manage the padding around these menus

File: src/styles/ia-bookmarks-list.js (hunk 1 above)
Comment (shaneriley) [reply src/styles/ia-bookmarks-list.js:49]: This is the width for the button containing the SVG icon, whose size will remain unchanged as the relative font size changes.

File: src/ia-bookmarks-list.js (hunk 3 above)
Comment (shaneriley) [reply src/ia-bookmarks-list.js]: That's true. I was thinking BookReader might have a listener on the document for updating the state of a given page's bookmark button, but that could be handled by the controller instead.

File: src/styles/ia-bookmarks-list.js (hunk 2 above)
Comment (shaneriley) [reply src/styles/ia-bookmarks-list.js]: Thanks! Added.

**Review by iisa — APPROVED** [review]
looks good! 🍰

**Pull request merged**